### PR TITLE
docs: stabilize integrations reference rendering

### PR DESCRIPTION
**Pull request by DyanGalih**
## Summary
This PR keeps the integrations docs rendering work from PR #2563 scoped to the PR-2563 branch.

## What changed
- Hardened integrations docs rendering helpers so markdown table output stays stable.
- Added safer escaping for markdown link text and repository URLs.
- Improved tests for the integrations reference table and community catalog docs.
- Kept the `integration search --markdown` CLI path aligned with the docs rendering tests.

## Why
The generated integrations reference was sensitive to ordering and markdown characters in labels, notes, and repository values. The tests now cover those edge cases so the docs output stays predictable.

## Impact
- More stable generated integrations docs.
- Better handling of special characters in table rows and links.
- Cleaner test coverage around the docs rendering paths.

## Validation
- `pytest tests/test_catalog_docs.py tests/test_community_catalog_docs.py -q`
- `python3 -m compileall -q src/specify_cli/__init__.py src/specify_cli/catalog_docs.py src/specify_cli/community_catalog_docs.py tests/test_catalog_docs.py tests/test_community_catalog_docs.py`

### CODE REVIEW

**Comment by mnriem** [comment]
Please address Copilot feedback

**Comment by DyanGalih** [comment]
I have addressed the Copilot feedback by introducing a `render_code_span` helper to safely handle strings containing backticks when generating the markdown tables. This fixes the issues with malformed tables for tags and IDs.

**Comment by DyanGalih** [comment]
I have also addressed the remaining feedback: `CliRunner` is now instantiated with `mix_stderr=False` to fix flaky tests, and `ROOT_DIR` is globally cached in `_assets.py` to avoid redundant path traversal. All threads are now resolved.

**Comment by DyanGalih** [comment]
I have also addressed the latest feedback by adding `zed` to the doc URLs map and updating `list_integrations_for_docs` to fall back to the integration install_url when an override is not present. All review threads are now resolved!

**Review by mnriem — CHANGES_REQUESTED** [review]
Please address Copilot feedback and fix test & lint errors

**Comment by DyanGalih** [comment]
I have addressed the remaining PR feedback. I updated `_format_tags()` in `community_catalog_docs.py` to correctly normalize newlines in tags using `render_cell()`. I also updated the relevant tests in `test_community_catalog_docs.py` to include coverage for tags with newlines. The local linters and tests all pass, and the changes have been pushed to the PR branch.

**Comment by mnriem** [comment]
Please address Copilot feedback and also address test & lint errors

File: src/specify_cli/catalog_docs.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+def escape_markdown_link_text(text: str) -> str:
+    """Escape characters that can break Markdown link text."""
+    return text.replace("[", "\\[").replace("]", "\\]")
+
+
+def render_code_span(value: str) -> str:
+    """Safely render a value as an inline markdown code span.
+    
+    Replaces internal backticks with single quotes to prevent breaking the code span.
+    """
```
[the remark this answers is not in the document]
Comment (DyanGalih) [reply src/specify_cli/catalog_docs.py]: Fixed the trailing whitespace in the docstring.

**Comment by mnriem** [comment]
Please address Copilot feedback and fix test & lint errors

File: src/specify_cli/catalog_docs.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    missing = sorted(registry_keys - set(INTEGRATION_DOC_URLS))
+    if missing and warn_on_missing:
+        import warnings
+        warnings.warn(
+            f"Integration(s) missing from INTEGRATION_DOC_URLS: "
+            f"{', '.join(missing)}. They will be included in the docs table "
+            "without documentation links. Add them to INTEGRATION_DOC_URLS in "
+            "catalog_docs.py if a link should be available.",
+            stacklevel=2
+        )
```
[the remark this answers is not in the document]
Comment (DyanGalih) [reply src/specify_cli/catalog_docs.py]: Fixed in this branch: the warning call now uses the trailing-comma style used elsewhere in the file.

File: src/specify_cli/integrations/_query_commands.py [hunk 3]
```diff
@@ -265,8 +265,33 @@ def integration_search(
...
+                "Warning: --markdown outputs the full built-in integrations table "
+                "and ignores query/--tag/--author filters.",
+                err=True,
+            )
+        from ..catalog_docs import render_integrations_table
+        try:
+            typer.echo(render_integrations_table(), nl=False)
+        except (FileNotFoundError, ValueError) as exc:
+            typer.echo(f"Error rendering integrations table: {exc}", err=True)
+            raise typer.Exit(code=1)
```
[the remark this answers is not in the document]
Comment (DyanGalih) [reply src/specify_cli/integrations/_query_commands.py]: Fixed in this branch: the CLI error path now uses `typer.Exit(1)` to match the rest of the file.

File: src/specify_cli/catalog_docs.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            stale_keys = sorted(
+                set(extra_in_urls + extra_in_labels + extra_in_notes)
+            )
+            warnings.warn(
+                f"Stale key(s) found in doc maps (no longer in registry): "
+                f"{', '.join(stale_keys)}. Consider removing them from "
+                "INTEGRATION_DOC_URLS, INTEGRATION_LABEL_OVERRIDES, and "
+                "INTEGRATION_NOTES.",
+                stacklevel=2
+            )
```
[the remark this answers is not in the document]
Comment (DyanGalih) [reply src/specify_cli/catalog_docs.py]: Fixed in this branch: the multiline warning call now keeps the trailing comma style used elsewhere in the file.

**Review by mnriem — CHANGES_REQUESTED** [review]
Please address Copilot feedback. And please resolve test & lint errors

**Comment by DyanGalih** [comment]
Addressed the latest feedback on the integrations docs rendering path.

What changed:
- Updated the `list_integrations_for_docs` docstring and warning text to match the actual `install_url` fallback behavior.
- Added regression coverage for the missing-doc-url fallback path.
- Kept the community catalog escaping fixes in place for tags and IDs.

Validation:
- `PYTHONPATH=src uv run pytest tests/test_catalog_docs.py tests/test_community_catalog_docs.py`
- `PYTHONPATH=src uv run ruff check .`

I also resolved the remaining review threads that were already addressed.

**Comment by mnriem** [comment]
Please address Copilot feedback and test & lint errors
